torch/_dynamo/symbolic_convert.py

```diff
@@ -949,6 +949,61 @@ def COMPARE_OP(self, inst):
                     **options,
                 )
             )
+        elif (
+            isinstance(left, BaseListVariable)
+            and all(
+                isinstance(l, (ConstantVariable, DynamicShapeVariable))
+                for l in left.items
+            )
+            and isinstance(right, BaseListVariable)
+            and all(
+                isinstance(r, (ConstantVariable, DynamicShapeVariable))
+                for r in right.items
+            )
+            and len(left.items) == len(right.items)
+            and op in supported_tensors
```

Comment:
I think you support too many ops here. If I do an inequality it needs to be treated lexicographically. But I don't see any such logic. You probably didn't care about support inequalities, but they are technically supported here.

Reply:
Good point. I guess I'll just reduce the number of ops I support here. Although I'm still curious, similar to the above comment - is there a way to get the different types of list equalities to be handled more automagically? Or are we stuck re-implementing stuff like that in dynamo.

Reply:
Anything that is morally implemented in CPython C code must be reimplemented. That being said, we can always create local abstractions to reduce duplication, etc.